app/src/main/java/net/aiscope/gdd_app/ui/metadata/MetadataPresenter.kt

```diff
@@ -53,12 +55,13 @@ class MetadataPresenter @Inject constructor(
         view.showInvalidFormError()
     }
 
-    suspend fun save(smearTypeId: Int, speciesValue: String) {
+    suspend fun save(smearTypeId: Int, speciesValue: String, comments: String?) {
```

Comment:
This method is never called with a `null` value for the `comments` parameter, why make it nullable?

Reply:
I'm not sure if in the `MetadataActivity` when we save and call `binding.metadataCommentsInput.text.toString()` if we could have a null pointer exception given that `.text` returns an `Editable!`. If that's the case I think it may make sense to leave the type as `String?`.

Reply:
Just found this: https://stackoverflow.com/questions/49561058/does-textview-gettext-ever-returns-null
If you want to be super defensive, you can default it to an empty string if for some weird reason the text is null, but I don't think that it's worth it.